File: refactor.dart
import 'dart:io';

void main() {
  final dir = Directory(r'c:\Users\imam\source\repos\imam8822\societyapp\lib');
  final files = dir.listSync(recursive: true).whereType<File>().where((f) => f.path.endsWith('.dart'));

  for (final file in files) {
    if (file.path.contains('app_utils.dart')) continue;

    var content = file.readAsStringSync();
    var original = content;

    // Standard single line catches
    content = content.replaceAll(
      "ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text(e.toString())));",
      "AppUtils.showError(context, e.toString());"
    );
    content = content.replaceAll(
      "ScaffoldMessenger.of(this.context).showSnackBar(SnackBar(content: Text(e.toString())));",
      "AppUtils.showError(this.context, e.toString());"
    );
    content = content.replaceAll(
      "ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text(apiError(e))));",
      "AppUtils.showError(context, apiError(e));"
    );
    content = content.replaceAll(
      "ScaffoldMessenger.of(context)\n            .showSnackBar(SnackBar(content: Text(apiError(e))));",
      "AppUtils.showError(context, apiError(e));"
    );
    content = content.replaceAll(
      "ScaffoldMessenger.of(context).showSnackBar(\n            const SnackBar(content: Text('Member added successfully!')));",
      "AppUtils.showSuccess(context, 'Member added successfully!');"
    );
    content = content.replaceAll(
      "ScaffoldMessenger.of(context).showSnackBar(\n            const SnackBar(content: Text('Settings saved!')));",
      "AppUtils.showSuccess(context, 'Settings saved!');"
    );
    content = content.replaceAll(
      "ScaffoldMessenger.of(context).showSnackBar(\n            const SnackBar(content: Text('Member updated successfully')));",
      "AppUtils.showSuccess(context, 'Member updated successfully');"
    );
    content = content.replaceAll(
      "ScaffoldMessenger.of(context).showSnackBar(\n            SnackBar(content: Text(e.toString())));",
      "AppUtils.showError(context, e.toString());"
    );

    // Multiline specific blocks
    content = content.replaceAll(
      "ScaffoldMessenger.of(context).showSnackBar(\n                        SnackBar(content: Text('Error: \$e')),\n                      );",
      "AppUtils.showError(context, 'Error: \$e');"
    );

    content = content.replaceAll(
      "ScaffoldMessenger.of(context).showSnackBar(\n        SnackBar(content: const Text('Amount must be greater than zero'), backgroundColor: context.colors.error),\n      );",
      "AppUtils.showError(context, 'Amount must be greater than zero');"
    );
    content = content.replaceAll(
      "ScaffoldMessenger.of(context).showSnackBar(\n          SnackBar(\n            content: const Text('Ledger adjustment recorded successfully'),\n            backgroundColor: context.colors.primary,\n          ),\n        );",
      "AppUtils.showSuccess(context, 'Ledger adjustment recorded successfully');"
    );
    content = content.replaceAll(
      "ScaffoldMessenger.of(context).showSnackBar(\n          SnackBar(content: Text(apiError(e)), backgroundColor: context.colors.error),\n        );",
      "AppUtils.showError(context, apiError(e));"
    );

    // If there were any replacements, add import
    if (content != original) {
      if (content.contains('AppUtils.show') && !content.contains('core/app_utils.dart')) {
        final importIdx = content.indexOf('import ');
        if (importIdx != -1) {
          final endImport = content.indexOf('\n', importIdx);
          content = content.substring(0, endImport + 1) + "import 'package:societyapp/core/app_utils.dart';\n" + content.substring(endImport + 1);
        }
      }
      file.writeAsStringSync(content);
      print('Updated \${file.path}');
    }
  }
}

File: refactor.py
import os
import re

lib_dir = r"c:\Users\imam\source\repos\imam8822\societyapp\lib"

# Matches: ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text(something)));
# with optional newlines and indentation
pattern_error = re.compile(r'ScaffoldMessenger\.of\((.*?context.*?)\)\s*\.\s*showSnackBar\(\s*SnackBar\(\s*content:\s*Text\(([^)]+)\)\s*(?:,\s*backgroundColor:\s*[^)]+)?\)\s*\);?', re.MULTILINE | re.DOTALL)

# Same for success cases (we'll look for positive sounding strings like "successfully")
# Wait, some have specific colors or just simple text.

for root, dirs, files in os.walk(lib_dir):
    for file in files:
        if file.endswith(".dart") and file != "app_utils.dart":
            path = os.path.join(root, file)
            with open(path, "r", encoding="utf-8") as f:
                content = f.read()

            original = content
            
            def repl(m):
                ctx = m.group(1).strip()
                text = m.group(2).strip()
                # If text contains "success" case-insensitive, use showSuccess
                if "success" in text.lower():
                    return f"AppUtils.showSuccess({ctx}, {text});"
                else:
                    return f"AppUtils.showError({ctx}, {text});"
            
            content = pattern_error.sub(repl, content)
            
            # Also handle shared_widgets.dart specific snackbar variable
            content = content.replace("ScaffoldMessenger.of(context)\n      ..hideCurrentSnackBar()\n      ..showSnackBar(snackBar);", "")
            
            if content != original:
                # Add import if missing
                if "AppUtils.show" in content and "core/app_utils.dart" not in content:
                    # insert after first import
                    import_idx = content.find("import ")
                    if import_idx != -1:
                        end_import = content.find("\n", import_idx)
                        content = content[:end_import+1] + "import 'package:societyapp/core/app_utils.dart';\n" + content[end_import+1:]
                
                with open(path, "w", encoding="utf-8") as f:
                    f.write(content)
                print(f"Updated {file}")
